pkg/api/vlabs/types.go

```diff
@@ -200,17 +201,19 @@ type KubernetesConfig struct {
 
 // MasterProfile represents the definition of the master cluster
 type MasterProfile struct {
-	Count                    int    `json:"count" validate:"required,eq=1|eq=3|eq=5"`
-	DNSPrefix                string `json:"dnsPrefix" validate:"required"`
-	VMSize                   string `json:"vmSize" validate:"required"`
-	OSDiskSizeGB             int    `json:"osDiskSizeGB,omitempty" validate:"min=0,max=1023"`
-	VnetSubnetID             string `json:"vnetSubnetID,omitempty"`
-	VnetCidr                 string `json:"vnetCidr,omitempty"`
-	FirstConsecutiveStaticIP string `json:"firstConsecutiveStaticIP,omitempty"`
-	IPAddressCount           int    `json:"ipAddressCount,omitempty" validate:"min=0,max=256"`
-	StorageProfile           string `json:"storageProfile,omitempty" validate:"eq=StorageAccount|eq=ManagedDisks|len=0"`
-	HTTPSourceAddressPrefix  string `json:"HTTPSourceAddressPrefix,omitempty"`
-	OAuthEnabled             bool   `json:"oauthEnabled"`
+	Count                    int         `json:"count" validate:"required,eq=1|eq=3|eq=5"`
+	DNSPrefix                string      `json:"dnsPrefix" validate:"required"`
+	VMSize                   string      `json:"vmSize" validate:"required"`
+	OSDiskSizeGB             int         `json:"osDiskSizeGB,omitempty" validate:"min=0,max=1023"`
+	VnetSubnetID             string      `json:"vnetSubnetID,omitempty"`
+	VnetCidr                 string      `json:"vnetCidr,omitempty"`
+	FirstConsecutiveStaticIP string      `json:"firstConsecutiveStaticIP,omitempty"`
+	IPAddressCount           int         `json:"ipAddressCount,omitempty" validate:"min=0,max=256"`
+	StorageProfile           string      `json:"storageProfile,omitempty" validate:"eq=StorageAccount|eq=ManagedDisks|len=0"`
+	HTTPSourceAddressPrefix  string      `json:"HTTPSourceAddressPrefix,omitempty"`
+	OAuthEnabled             bool        `json:"oauthEnabled"`
+	PreprovisionExtension    *Extension  `json:"preprovisionExtension"`
```

Comment:
nit: change to `PreProvisionExtension` (capital `P` in `Provision`)

(and `preProvisionExtension` for the json representation)